modules/telemetry/telemetrymodule.cpp

```diff
@@ -0,0 +1,314 @@
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/telemetry/telemetrymodule.h>
+
+#include <modules/telemetry/include/general/anglemodetelemetry.h>
+#include <modules/telemetry/include/general/cameratelemetry.h>
+#include <modules/telemetry/include/general/focustelemetry.h>
+#include <modules/telemetry/include/general/nodestelemetry.h>
+#include <modules/telemetry/include/general/timetelemetry.h>
+#include <modules/telemetry/include/specific/planetscomparesonification.h>
+#include <modules/telemetry/include/specific/planetsoverviewsonification.h>
+#include <modules/telemetry/include/specific/planetssonification.h>
+#include <ghoul/logging/logmanager.h>
+#include <openspace/camera/camera.h>
+#include <openspace/documentation/documentation.h>
+#include <openspace/engine/globals.h>
+#include <openspace/engine/globalscallbacks.h>
+#include <openspace/engine/windowdelegate.h>
+#include <openspace/rendering/renderengine.h>
+#include <openspace/scene/scene.h>
+
+namespace {
+    constexpr std::string_view _loggerCat = "TelemetryModule";
+
+    // The default OSC reciver is SuperCollider with these default values.
+    // However, the user can define any reciver in the openspace.cfg file as the
+    // ModuleConfiguration for the Telemetry module.
+    constexpr std::string_view DefaultSuperColliderIp = "127.0.0.1";
+    constexpr int DefaultSuperColliderPort = 57120;
+
+    constexpr openspace::properties::Property::PropertyInfo EnabledInfo = {
+        "Enabled",
+        "Enabled",
+        "Enable or disable all gathering of telemetry information"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo IpAddressInfo = {
+        "IpAddress",
+        "Ip address",
+        "The network ip address that the telemetry osc messages is sent to"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo PortInfo = {
+        "Port",
+        "Port",
+        "The network port that the telemetry osc messages is sent to"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo AngleCalculationModeInfo = {
+        "AngleCalculationMode",
+        "Angle Calculation Mode",
+        "This setting changes the method to calculate any angles in the telemetries. "
```

Comment:
```suggestion
        "This setting changes the method for how the angles in the telemetries are computed. "
```

Not completely sure on this phrasing. Is "in the telemetries" correct here? 🙃

Reply:
Is this small change any better? (Copilot thought so...)
```
"This setting changes the method used to calculate any angles in the "
"telemetries.
```